### PR TITLE
Making WarBot a webpage - adding flask

### PR DESCRIPTION
**Pull request by mafilipp**
Implements a basic webpage to select the desired world and play the game. The output of the game is displayed.

### CODE REVIEW

**Comment by mafilipp** [comment]
@RiLeone  I thought that for the world selection it would be nice to have a drop down. How would you suggest me to integrate it with WorldSelector? Imo I could just have a mapping
name_world -> file_to_select.json
What do you think?

File: app.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import HistoricStatistician as hs
+
+
+def main():
+    """Run the full game."""
+
+    print(__doc__)
+    options = AuxiliaryTools.parse_args()
+
+    world_file = './worlds/Debugland/states.json'
```
Comment (RiLeone) [inline app.py:25]: For using the drop-down  menu (which I really like, in particular in the web-oriented approach) I would use the `AuxiliaryTools.list_worlds()` function here or just before generating the dropdown to get all available worlds (each world then is required to have a `states.josn` file in any case to work).

File: app.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+app = Flask(__name__)
+
+
+@app.route("/")
+def home():
+    text = "Hello! welcome to warbot"
+    bold_text = "Whup Whup!"
+    return render_template('home.html', text=text, bold_text=bold_text)
```
Comment (RiLeone) [inline app.py:37]: I peresoanlly prefer packing all kwargs in a dictionary and then pass the **dict as single argument to `render_template()`, but this is purely personal preference :)

File: templates/home.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{% extends "base.html" %}
+{% block title %}General Page{% endblock %}
+
+{% block content %}
+    {{text}}
+    <h1>{{bold_text}}</h1>
```
Comment (RiLeone) [inline templates/home.html:6]: I would have called it `title` instead of `bold_text`, but again, this is only personal preference.

**Review by RiLeone — CHANGES_REQUESTED** [review]
IMO it is better if we wait for your suggestions (drop-down, rpoper text formatting, replacement of placehodelrs with "meaningful" text) to be implemented before merging, so that the merged version already has all basic features we want and does not contain any palceholders anymore. Would that be ok for you? Also, could you also add a basic shell script for running the flask app (in`x/`)? I know, right now it would be a single-line file, but later on, if we have some args to pass we already have a flexiblesolution in place.

File: app.py (hunk 2 above)
Comment (mafilipp) [reply app.py:37]: This is actually what is happening. Indeed, the function offered by flask API is:
`def render_template(template_name_or_list, **context)`

Comment (mafilipp) [reply app.py:37]: In general, I agree definitely with you, it keeps the code cleaner

**Comment by mafilipp** [comment]
I updated my PR @RiLeone. IMO like that is finished and could be merged, after your review of course :)
Let me know what do you think.
About your suggestion on the script to start flask, I can imagine you mean something like [that](https://stackoverflow.com/a/41264536). I am not sure though if we should give the possibility of command-line arguments. imo the app should be started just once, and everything that we can change on the webpage should be configurable on the webpage.. what do you think?

Thank you for the review

File: app.py (hunk 1 above)
Comment (mafilipp) [reply app.py]: Thank you for the suggestion

**Comment by RiLeone** [comment]
> I updated my PR @RiLeone. IMO like that is finished and could be merged, after your review of course :)
> Let me know what do you think.
> About your suggestion on the script to start flask, I can imagine you mean something like [that](https://stackoverflow.com/a/41264536). I am not sure though if we should give the possibility of command-line arguments. imo the app should be started just once, and everything that we can change on the webpage should be configurable on the webpage.. what do you think?
> 
> Thank you for the review

I was more thinking of a shell-script (probably a single liner), but you are right. Let's keep it like it is now :D

**Review by RiLeone — COMMENTED** [review]
All good

**Pull request merged**